python/instrumentation/openinference-instrumentation-langchain/src/openinference/instrumentation/langchain/__init__.py

```diff
@@ -64,6 +64,19 @@ def _uninstrument(self, **kwargs: Any) -> None:
     def get_span(self, run_id: UUID) -> Optional[Span]:
         return self._tracer.get_span(run_id) if self._tracer else None
 
+    def get_root_chain_spans(self, run_id: UUID) -> Optional[List[Span]]:
+        span = self.get_span(run_id)
+        root_chain_spans = []
+        while span and span.get_span_context().is_valid:
+            span_id = span.get_span_context().span_id
+            tracer = self._tracer
+            assert tracer
+            if span_id in tracer._root_span_ids:
```

Comment:
As mentioned earlier, in this function you could use the existing `tracer.run_map` from the base `Tracer` to simplify your implementation, rather than accessing private attributes in this scope. Since `tracer.run_map` already provides all the necessary functionality, there’s no need to re-implement everything from scratch: the parts highlighted by arrows in the screenshot below would give you the [`parent_run_id`](https://github.com/langchain-ai/langsmith-sdk/blob/5e21109e1cd70ee19db29941c128223ef360a40b/python/langsmith/schemas.py#L254) and [`run_type`](https://github.com/langchain-ai/langsmith-sdk/blob/5e21109e1cd70ee19db29941c128223ef360a40b/python/langsmith/schemas.py#L225) that you're using for your tree traversal algorithm.

<img width="739" alt="Screenshot 2024-10-16 at 8 11 51 AM" src="https://github.com/user-attachments/assets/7f6f783e-9f2e-442a-a1e0-69085e560625">